### PR TITLE
Product template library

### PR DESCRIPTION
**Pull request by MyagmarsurenM**
Template UI in Settings

Screen Shot 2022-02-18 at 20 27 39

Add new templates

Screen Shot 2022-02-18 at 20 27 47

Left sidebar on template UI

Screen Shot 2022-02-18 at 20 27 55

GrowthHacks template

Screen Shot 2022-02-18 at 20 28 09

ChatResponse template

Screen Shot 2022-02-18 at 20 28 19

Email template

Screen Shot 2022-02-18 at 20 28 29

Add new product template on salesPipeline

Screen Shot 2022-02-18 at 20 28 54

Screen Shot 2022-02-18 at 20 32 03

Save as product template on salesPipeline

Screen Shot 2022-02-18 at 20 32 13

Template icon on settings

Screen Shot 2022-02-18 at 20 32 29

### CODE REVIEW

File: api/src/data/resolvers/productCategory.ts [hunk 1]
```diff
@@ -18,4 +18,4 @@ export default {
       status: { $ne: PRODUCT_STATUSES.DELETED }
     });
   }
-};
+};
```
Comment (munkhsaikhan) [inline api/src/data/resolvers/productCategory.ts:21]: !!!

File: ui/src/modules/settings/growthHacks/components/TemplateList.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import dayjs from 'dayjs';
 import Button from 'modules/common/components/Button';
-import HeaderDescription from 'modules/common/components/HeaderDescription';
+// import HeaderDescription from 'modules/common/components/HeaderDescription';
```
Comment (munkhsaikhan) [inline ui/src/modules/settings/growthHacks/components/TemplateList.tsx:3]: delete comment

File: ui/yarn.lock [hunk 3]
```diff
@@ -6076,7 +6076,7 @@ fb-watchman@^2.0.0:
   dependencies:
     bser "^2.0.0"
 
-fbjs@^0.8.0, fbjs@^0.8.1, fbjs@^0.8.15, fbjs@^0.8.16:
+fbjs@^0.8.0, fbjs@^0.8.15, fbjs@^0.8.16:
```
Comment (munkhsaikhan) [inline ui/yarn.lock:6079]: why package.json unchanged, yarn.lock changed?

File: ui/src/modules/settings/emailTemplates/constants.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export const EMAIL_TEMPLATE_STATUSES = {
+  ACTIVE: 'active',
+  ARCHIVED: 'archived',
+  ALL: ['active', 'archived']
+};
+
+export const EMAIL_TEMPLATE_TIPTEXT = {
```
Comment (munkhsaikhan) [inline ui/src/modules/settings/emailTemplates/constants.ts:7]: check need

File: ui/src/modules/settings/growthHacks/components/TemplateList.tsx (hunk 2 above)
Comment (MyagmarsurenM) [reply ui/src/modules/settings/growthHacks/components/TemplateList.tsx]: deleted comment

File: api/src/data/resolvers/productCategory.ts (hunk 1 above)
Comment (MyagmarsurenM) [reply api/src/data/resolvers/productCategory.ts]: discarded the change

File: ui/yarn.lock (hunk 3 above)
Comment (MyagmarsurenM) [reply ui/yarn.lock]: discarded the change

File: ui/src/modules/deals/components/product/ProductForm.tsx [hunk 5]
```diff
@@ -87,6 +93,73 @@ class ProductForm extends React.Component<Props, State> {
...
+      discountPercent: discountT || 0,
+      amount: unitPrice - discountAmount || 0,
+      currency,
+      tickUsed: true,
+      product: product ? product : ({} as IProduct)
+    });
+
+    this.updateTotal(productsData);
+    onChangeProductsData(productsData);
+    // });
```
Comment (munkhsaikhan) [inline ui/src/modules/deals/components/product/ProductForm.tsx:128]: comment

File: ui/src/modules/settings/template/containers/product/SelectProducts.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import SelectProducts from 'erxes-ui/lib/products/containers/SelectProducts';
```
Comment (munkhsaikhan) [inline ui/src/modules/settings/template/containers/product/SelectProducts.tsx:1]: erxes-ui

**Pull request merged**